src/axolotl/utils/data.py

```diff
@@ -239,8 +251,9 @@ def for_d_in_datasets(dataset_configs):
                 raise ValueError("unhandled dataset load")
             # support for using a subset of the data
             if config_dataset.shards:
-                if "train" in ds:
-                    ds = ds.shuffle(seed=seed)["train"].shard(
+                shard_split = config_dataset.split if config_dataset.split else split
```

Comment:
Maybe we can refactor this code to split first (using the code around L270-275), then shard after?